### PR TITLE
test: Run mempool_packages.py with MiniWallet

### PR DESCRIPTION
**Pull request by maflcko**
This allows to run the test even when no wallet is compiled in.

Also, it is a lot nicer to read now.

### CODE REVIEW

**Comment by 0xB10C** [comment]
Concept ACK

**Comment by theStack** [comment]
Concept ACK

File: test/functional/mempool_packages.py [hunk 1]
```diff
@@ -29,7 +27,7 @@
 
 class MempoolPackagesTest(BitcoinTestFramework):
     def add_options(self, parser):
-        self.add_wallet_options(parser)
+        self.add_wallet_options(parser, legacy=False)
```
Comment (glozow) [inline test/functional/mempool_packages.py:30]: question, why `legacy=False`?

**Review by glozow — COMMENTED** [review]
utACK fa834fa3d5501214e6e98b57d046b85d6e8300c9, looks correct to me. one nit

File: test/functional/mempool_packages.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/mempool_packages.py]: Thanks, dropped for now

File: test/functional/test_framework/wallet.py [hunk 2]
```diff
@@ -330,38 +330,31 @@ def sendrawtransaction(self, *, from_node, tx_hex, maxfeerate=0, **kwargs):
...
 
-        return {"chain_hex": chain_hex, "chain_txns" : chain_txns}
+        return chain
 
-    def send_self_transfer_chain(self, *, from_node, chain_length, utxo_to_spend=None):
+    def send_self_transfer_chain(self, *, from_node, **kwargs):
         """Create and send a "chain" of chain_length transactions. The nth transaction in
         the chain is a child of the n-1th transaction and parent of the n+1th transaction.
 
         Returns the chaintip (nth) utxo
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply test/functional/test_framework/wallet.py]: thanks, fixed doc

**Comment by MarcoFalke** [comment]
rebased

**Comment by glozow** [comment]
reACK fa6b402

**Pull request merged**